src/platform/datapath_winuser.c

```diff
@@ -606,14 +796,21 @@ QuicDataPathInitialize(
     }
 
     RtlZeroMemory(Datapath, DatapathLength);
-    Datapath->RecvHandler = RecvCallback;
-    Datapath->UnreachableHandler = UnreachableCallback;
+    if (UdpCallbacks) {
+        Datapath->UdpHandlers = *UdpCallbacks;
```

Comment:
This requires the upper layer (QUIC core) to keep this structure alive for the duration of the datapath layer, which kind of breaks the layering abstraction between them.  Shouldn't this copy the structure/pointers locally?

Reply:
The deref will copy the struct, keeping a local copy of the pointers.

Reply:
Ah, I (incorrectly) assumed it was a pointer-pointer. You're right.